package/MDAnalysis/core/selection.py

```diff
@@ -679,12 +679,10 @@ def _apply(self, group):
             raise ValueError(f"{self.pattern!r} is not a valid SMARTS query")
         mol = group.convert_to("RDKIT", **self.rdkit_kwargs)
         matches = mol.GetSubstructMatches(pattern, useChirality=True)
-        # convert rdkit indices to mdanalysis'
-        indices = [
-            mol.GetAtomWithIdx(idx).GetIntProp("_MDAnalysis_index")
-            for match in matches for idx in match]
+        # flatten all matches and remove duplicated indices
+        indices = np.unique([idx for match in matches for idx in match])
         # create boolean mask for atoms based on index
-        mask = np.in1d(range(group.n_atoms), np.unique(indices))
+        mask = np.in1d(range(group.n_atoms), indices)
```

Comment:
LGTM. this works because the entire group is converted and therefore RDKit index should == MDAnalysis_index (relative to `group`).